### PR TITLE
[DBEX] add mental health conditions screener page

### PR DESCRIPTION
**Pull request by freeheeling**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
  - Create the first (screener) page of the new, digitized Form 0781 flow ([page design](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=9250-79794&t=pxzExlOnHcv68GkK-4))
- _(If bug, how to reproduce)_
  - n/a
- _(What is the solution, why is this the solution)_
  - Create a new flow and pages to align the digital form with the [VA Form 21-0781](https://www.vba.va.gov/pubs/forms/VBA-21-0781-ARE.pdf) paper version
- _(Which team do you work for, does your team own the maintenance of this component?)_
  - Disability Benefits Experience Team 2 (dBeX Carbs 🥖), which owns maintenance of all files in this PR
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
  - `sync_modern_0781_flow` will be utilized for specific users until all who start a new form and claim a new disability will be routed through an alternative 0781 flow

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/96933
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/87438

## Testing done

- _Describe what the old behavior was prior to the change_
 - This page did not previously exist on the digital form
- _Describe the steps required to verify your changes are working as expected_
  1. enable `sync_modern_0781_flow` toggle 
  2. start a new claim
  3. the new page, Mental health conditions, should display after the 'Prisoner of war (POW)' page in the Disabilities chapter
  4. the new page should continue to the 'Additional disability benefits' page
- Describe the tests completed and the results
  - added unit tests
  - manually tested

## Screenshots

|                  | Mobile | Desktop |
| --------- | ------- | -------- |
| collapsed | Screenshot 2024-11-12 at 3 24 49 PM | Screenshot 2024-11-12 at 3 23 11 PM |
| expanded | Screenshot 2024-11-12 at 3 25 37 PMScreenshot 2024-11-12 at 3 25 57 PMScreenshot 2024-11-12 at 3 26 24 PM | Screenshot 2024-11-12 at 3 22 46 PM |
| invalid | Screenshot 2024-11-12 at 3 24 24 PM | Screenshot 2024-11-12 at 3 23 34 PM |

## What areas of the site does it impact?

- 526 EZ

### Quality Assurance & Testing

- [x] I ~fixed~|~updated~|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) ~has been performed~ (linked page does not exist)

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [X] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by aurora-a-k-a-lightning — DISMISSED** [review]

**Review by kylesoskin — DISMISSED** [review]
Looks good to me. 

It says in the description it will use flipper `sync_modern_0781_flow`, but I see no reference to that in the PR code. Is all this code just prep code, creating the pages, but not actually making anything routing to them yet? And that will come later, via that flipper?

**Comment by freeheeling** [comment]
@kylesoskin 

> It says in the description it will use flipper `sync_modern_0781_flow`, but I see no reference to that in the PR code. Is all this code just prep code, creating the pages, but not actually making anything routing to them yet? And that will come later, via that flipper?

See `depends: formData => showMentalHealthPages(formData)` in src/applications/disability-benefits/all-claims/config/form.js. The `showMentalHealthPages` function evaluates whether the `syncModern0781Flow` attribute in the `formData` is true, in addition to the claim being new and a new, unrated condition being claimed.

**Review by harshilmoradia — DISMISSED** [review]

File: src/applications/disability-benefits/all-claims/content/mentalHealth.jsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+          You experienced or witnessed friendly fire that occurred on a gunnery
+          range during a training mission
+        </li>
+      </ul>
+    </va-accordion-item>
+  </va-accordion>
+);
+
+export const noneAndConditionError =
+  "If you're not claiming any mental health conditions related to a traumatic event, unselect the other options you selected";
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/content/mentalHealth.jsx:88]: use [curly apostrophe](https://design.va.gov/content-style-guide/punctuation#apostrophes)

```suggestion
 'If you’re not claiming any mental health conditions related to a traumatic event, unselect the other options you selected';
```

File: src/applications/disability-benefits/all-claims/content/mentalHealth.jsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  <va-accordion open-single>
+    <va-accordion-item
+      header="Examples of traumatic events"
+      class="vads-u-margin-y--3"
+      id="first"
+      bordered
+    >
+      <h4 className="vads-u-margin-top--0">
+        Traumatic events related to combat
+      </h4>
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/content/mentalHealth.jsx:27]: Looks like we're skipping heading level here 2 -> 4
image

File: src/applications/disability-benefits/all-claims/content/mentalHealth.jsx (hunk 1 above)
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/content/mentalHealth.jsx]: Corrected in [6eba54f](https://github.com/department-of-veterans-affairs/vets-website/pull/32993/commits/6eba54f875309c70a4cce8918c3b97a6e4b710b6)

Screenshot 2024-11-19 at 11 16 39 AM

File: src/applications/disability-benefits/all-claims/content/mentalHealth.jsx (hunk 2 above)
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/content/mentalHealth.jsx:23]: Please review the latest commit, [6eba54f](https://github.com/department-of-veterans-affairs/vets-website/pull/32993/commits/6eba54f875309c70a4cce8918c3b97a6e4b710b6), that aims to address proper heading levels.

Screenshot 2024-11-19 at 11 11 43 AM

**Review by christinec-fftc — APPROVED** [review]

**Pull request merged**